src/main/java/seedu/address/commons/util/FileUtil.java

```diff
@@ -123,6 +124,27 @@ public static boolean hasExtension(Path filePath, String[] allowedExtensions) {
                 .reduce(false, (x, y) -> x || y);
     }
 
+    /**
+     * Checks if the content of the file is prefixed with any magic number from a given set.
+     * @throws IOException If the file cannot be read.
+     */
+    public static boolean hasMagicNumber(Path filePath, Set<Byte[]> allowedMagicNumber) throws IOException {
+        byte[] bytes = Files.readAllBytes(filePath);
+        Byte current;
+        for (Byte[] magicNumber : allowedMagicNumber) {
+            for (int i = 0; i < magicNumber.length; i++) {
```

Comment:
Better add in a comment saying that null bytes in magic numbers are like wildcards and can take on any values